File: solution/028ImplementStrStr/028ImplementStrStr.go
package main

import (
	"fmt"
)

func strStr(haystack string, needle string) int {
	j := 0
	for j = 0; j < len(haystack)-len(needle)+1; j++ {
		i := 0
		for i = 0; i < len(needle); i++ {
			if haystack[j+i] != needle[i] {
				break
			}
		}
		if i == len(needle) {
			return j
		}
	}
	return -1
}

func main() {
	fmt.Printf("%v\n", strStr("abcdef", "def"))
	fmt.Printf("%v\n", strStr("abcdef", "daf"))
	fmt.Printf("%v\n", strStr("abcdef", "ab"))
}

File: solution/028ImplementStrStr/028ImplementStrStr.rs
impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        let n1 = haystack.len();
        let n2 = needle.len();
        let h = haystack.as_bytes();
        let n = needle.as_bytes();
        for i in 0..(n1-n2+1) as i32 {
            let mut j:usize = 0;
            while j < n2 {
                if h[(i as usize)+j] != n[j] {
                    break;
                }    
                j += 1;
            }  
            if j == n2 {
                return i;
            }
        }
        -1
    }
}
